tests/bluetooth/bsim_bt/bsim_test_audio/src/ias_test.c

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (c) 2022 Codecoup
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdint.h>
+#ifdef CONFIG_BT_IAS
+
+#include <stddef.h>
+#include <errno.h>
+#include <zephyr/types.h>
+
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/bluetooth/hci.h>
+#include <zephyr/bluetooth/conn.h>
+#include <zephyr/bluetooth/uuid.h>
+#include <zephyr/bluetooth/gatt.h>
+
+#include <zephyr/bluetooth/services/ias.h>
+
+#include "common.h"
+
+extern enum bst_result_t bst_result;
+
+CREATE_FLAG(g_is_connected);
+CREATE_FLAG(g_high_alert_received);
+CREATE_FLAG(g_mild_alert_received);
+CREATE_FLAG(g_stop_alert_received);
+
+static void high_alert_cb(void)
+{
+	SET_FLAG(g_high_alert_received);
+}
+
+static void mild_alert_cb(void)
+{
+	SET_FLAG(g_mild_alert_received);
+}
+
+static void no_alert_cb(void)
+{
+	SET_FLAG(g_stop_alert_received);
+}
+
+BT_IAS_CB_DEFINE(ias_callbacks) = {
+	.high_alert = high_alert_cb,
+	.mild_alert = mild_alert_cb,
+	.no_alert = no_alert_cb,
+};
+
+static void connected(struct bt_conn *conn, uint8_t err)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	if (err) {
+		FAIL("Failed to connect to %s (%u)\n", addr, err);
+		return;
+	}
+
+	printk("Connected to %s\n", addr);
+	default_conn = bt_conn_ref(conn);
+	g_is_connected = true;
+}
+
+static struct bt_conn_cb conn_callbacks = {
+	.connected = connected,
+	.disconnected = disconnected,
+};
+
+static void test_main(void)
+{
+	int err;
+
+	err = bt_enable(NULL);
+	if (err != 0) {
+		FAIL("Bluetooth init failed (err %d)\n", err);
+		return;
+	}
+
+	bt_conn_cb_register(&conn_callbacks);
+
+	err = bt_le_adv_start(BT_LE_ADV_CONN_NAME, ad, AD_SIZE, NULL, 0);
+	if (err) {
+		FAIL("Advertising failed to start (err %d)\n", err);
+		return;
+	}
+
+	SET_FLAG(g_stop_alert_received);
+	bt_ias_local_alert_stop();
+	WAIT_FOR_FLAG(g_stop_alert_received);
+	printk("Stop alert received from local stop\n");
+	UNSET_FLAG(g_stop_alert_received);
```

Comment:
Suggest to move `bt_ias_local_alert_stop` after `g_high_alert_received` to clear the alert, and then also verify the return value of `bt_ias_local_alert_stop`

What was the point of setting the `g_stop_alert_received` flag and immediately check if it is set afterwards?